Description: Cope with filesystems that do not keep track of free blocks/inodes
Author: Gilles Chehade <gilles@poolp.org>
Origin: Upstream
Bug-Upstream: https://github.com/poolpOrg/OpenSMTPD/issues/246
Bug-Debian: http://bugs.debian.org/723893
Applied-Upstream: https://github.com/poolpOrg/OpenSMTPD/commit/7145d4c5
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/smtpd/queue_fsqueue.c
+++ b/smtpd/queue_fsqueue.c
@@ -108,6 +108,19 @@
 		return 0;
 	}
 
+
+	/* XXX */
+	/*
+	 * Queue does not run as root so these can't be == 0.
+	 *
+	 * Also, <= 0 means "undefined for this filesystem",
+	 * the tests should be skipped. Remaining code can
+	 * cope with shortage anyway...
+	 *
+	 */
+	if (buf.f_bfree <= 0 || buf.f_ffree <= 0)
+		return 1;
+
 	used = buf.f_blocks - buf.f_bfree;
 	total = buf.f_bavail + used;
 	if (total != 0)
